server/src/main/java/keywhiz/service/resources/automation/v2/BackfillRowHmacResource.java

```diff
@@ -38,11 +50,65 @@
 
   private final DSLContext jooq;
   private final RowHmacGenerator rowHmacGenerator;
+  private final AuditLog auditLog;
 
   @Inject
-  public BackfillRowHmacResource(DSLContext jooq, RowHmacGenerator rowHmacGenerator) {
+  public BackfillRowHmacResource(
+      DSLContext jooq,
+      RowHmacGenerator rowHmacGenerator,
+      AuditLog auditLog) {
     this.jooq = jooq;
     this.rowHmacGenerator = rowHmacGenerator;
+    this.auditLog = auditLog;
+  }
+
+  /**
+   * Backfill row_hmac for a specific secret.
+   */
+  @Timed @ExceptionMetered
+  @Path("secret/name/{secretName}")
+  @POST
+  @Consumes(APPLICATION_JSON)
+  @Produces(APPLICATION_JSON)
+  public void backfillSecretRowHmacByName(
+      @Auth AutomationClient automationClient,
+      @PathParam("secretName") String secretName,
+      @QueryParam("force") boolean force) {
+    Optional<SecretsRecord> maybeRow = jooq.selectFrom(SECRETS)
+        .where(SECRETS.NAME.eq(secretName))
+        .fetchOptional();
+
+    if (!maybeRow.isPresent()) {
+      throw new NotFoundException();
+    }
+
+    SecretsRecord row = maybeRow.get();
+    String oldHmac = row.getRowHmac();
+
+    if (oldHmac != null && !force) {
+      return;
+    }
```

Comment:
This is easy to overlook, but we're introducing a "force" parameter so that we can support two use cases:

1. An actual backfill (only populating the HMAC if it's missing) via force = false
2. Re-calculating/over-writing an existing HMAC via force = true

I wanted to do this so that we could follow the spirit of the existing APIs in this resource (which are strictly backfill-only) while also enabling our "repair" use case.